r/inst/NOTICE.txt

```diff
@@ -0,0 +1,8 @@
+Apache Arrow
+Copyright 2016-2019 The Apache Software Foundation
+
+This product includes software developed at
+The Apache Software Foundation (http://www.apache.org/).
+
+See https://github.com/apache/arrow/blob/master/NOTICE.txt for the complete
```

Comment:
@wesm is this legit to do? See https://github.com/apache/arrow/pull/4908/files#diff-3d1942ed064c42a43c8aef1fc78fe685R16 for why I put this here. Alternatively we could copy the full NOTICE.txt here, and/or do that at package build time, but this seemed less susceptible to bit rot/human error.

FWIW, there are very few (about 4, AFAICT) packages on CRAN that have ASF source header comments, and most also have this problem (referring to a NOTICE file but no NOTICE file is in the package). [Only one](https://github.com/cran/rkafkajars/blob/master/inst/NOTICE) of those packages includes a NOTICE file (in this same location), but the context is a bit different (the package is just a vehicle for delivering the Java .jar). The most similar package to ours, `sparkR` (with source contained inside the spark monorepo), does not contain a NOTICE file. However, CRAN editorial review of new package submissions has increased in recent months though, and responding "but what about `$PACKAGE` already on CRAN with the same issue?" is a nonstarter. I mention all this just to say that there is not much of a model we can follow here.

Reply:
Ideally NOTICE.txt would be included in the package

Reply:
Ok, done in https://github.com/apache/arrow/pull/4908/commits/0cec0f1e22e22ea497b69f89da0b71a29ce94718